### PR TITLE
Move Optics std to companion objects

### PR DESCRIPTION
**Pull request by nomisRev**
Like it was done for generated Optics in #825 this PR moves the Optics of the std to the companion objects of the respective data types.

### CODE REVIEW

**Comment by pakoito** [comment]
I see you've created new Api objects fro List and such. Rather than having a new one just for this library, it'd be better if we added an object in one of the core or data module, and add them as extension functions. That way it can be extended by all modules, and we centralise all the APIs.

**Comment by nomisRev** [comment]
@pakoito I made the changes 👍

File: modules/core/arrow-instances-data/src/main/kotlin/arrow/instances/list.kt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package arrow.instances
+
+object ListInstances
```
Comment (pakoito) [inline modules/core/arrow-instances-data/src/main/kotlin/arrow/instances/list.kt]: Could you please put this file in arrow-data instead? That way we don't need to depend on the instances package when adding other typeclasses to List.

File: modules/core/arrow-instances-data/src/main/kotlin/arrow/instances/set.kt [hunk 2]
```diff
@@ -1,73 +1,3 @@
...
-  override fun <A> Kind<ForSetK, A>.combineK(y: Kind<ForSetK, A>): SetK<A> =
-    fix().setCombineK(y)
-}
-
-object SetKContext : SetKFoldableInstance, SetKMonoidKInstance
-
-infix fun <A> ForSetK.Companion.extensions(f: SetKContext.() -> A): A =
-  f(SetKContext)
\ No newline at end of file
+object SetInstances
```
Comment (pakoito) [inline modules/core/arrow-instances-data/src/main/kotlin/arrow/instances/set.kt]: Same as ListInstances!

File: modules/core/arrow-instances-data/src/main/kotlin/arrow/instances/list.kt (hunk 1 above)
Comment (nomisRev) [reply modules/core/arrow-instances-data/src/main/kotlin/arrow/instances/list.kt]: Shouldn't these go to `arrow-core`?

Comment (pakoito) [reply modules/core/arrow-instances-data/src/main/kotlin/arrow/instances/list.kt]: It's cheap so why not :D

**Pull request merged**